modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -56,18 +84,11 @@ public function getNotificationsActivities(AccountInterface $account, array $sta
    *
    * @return int
    *   Number of remaining notifications.
-   *
-   * @throws \Drupal\Core\Entity\EntityStorageException
    */
-  public function markAllNotificationsAsSeen(AccountInterface $account) {
-
+  public function markAllNotificationsAsSeen(AccountInterface $account): int {
     // Retrieve all the activities referring this entity for this account.
-    $ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED]);
-
-    $activities = Activity::loadMultiple($ids);
-    foreach ($activities as $activity) {
-      $this->changeStatusOfActivity($activity, ACTIVITY_STATUS_SEEN);
-    }
+    $ids = $this->getNotificationIds($account);
+    $this->changeStatusOfActivity($ids, ACTIVITY_STATUS_SEEN);
```

Comment:
We should check if we have any IDs to parse here.

If we have no IDs the `changeStatusOfActivity` function will break.

Reply:
Agreed.